File: MarcenariaDotNet/Utils/NumberUtils.cs
using System.Globalization;
using System.Text.RegularExpressions;

namespace MarcenariaDotNet.Utils;

public static class NumberUtils 
{
  public const double TWO_PI = Math.PI * 2;
  public const double FOUR_PI = Math.PI * 4;

  public static double StrDouble(string str) 
  {
    str = Regex.Replace(str, "[^0-9,.]", "");
    str = str.Replace(',', '.');
    if (double.TryParse(str, out double value))
    {
      return value;
    } 

    return 0;
  }


  public static string DoubleToStr(double doub) 
  { 
    string str = doub.ToString("F2", CultureInfo.InvariantCulture);
    return str.Replace('.', ',');
  }

}

File: codes/java/marcenaria-java-activej/src/main/java/github/gmess/utils/NumberUtils.java
package github.gmess.utils;

public abstract class NumberUtils {
  public static final double TWO_PI = 2 * Math.PI;
  public static final double FOUR_PI = 4 * Math.PI;
  public static final double ONE_THIRD_PI = Math.PI / 3;

  public static final double stringToDouble(final String val) {
    return Double.valueOf(val.replace(",", ".").replaceAll("[^0-9.]", ""));
  }
}
